src/interceptor.ts

```diff
@@ -49,11 +49,16 @@ function createMetricsInterceptorProvider(
             collector.onStatusMetadataReceived(
               status as unknown as ServerStatus,
             );
+            collector.onAttemptComplete(status.code);
```

Comment:
As I mentioned in the meeting, I think it's probably better to only run onAttemptComplete when there's an error code, and let the onOperationComplete finalize everything on success

It doesn't have to be that way, but consider it and let me know what you think

Reply:
The only trouble is that at this time we don't know if the code is a retryable error or not. That is up to gax to decide. And we will run onOperationComplete when we get non-retryable errors or with 0 status code which includes a whole bunch of edge cases like max retries exceeded and timeout exceeded. If onOperationComplete is implicitly running onAttemptComplete and we only want onAttemptComplete to be run once then here we have to know if onOperationComplete is going to get called which is a hard thing to determine.

My suggestion for now is to leave this as is, but let me know if there is another alternative that addresses the timing concern.

Reply:
Think of it as two states: error or success. 
- If there's an error, we can end the attempt now, whether it retries or not. This current attempt is over.
- If it's successful, we can just leave the attempt running until we reach onOperationComplete, which closes them both

To me, this feels like a small change that makes the timing a bit more accurate, which is why I brought it up in the meeting. But honestly, it doesn't matter too much how we record that final successful attempt, so if you think it's confusing, we can just forget it and leave it as-is

Reply:
Sounds good. I think I misunderstood your suggestion before about implicitly ending the attempt if necessary in onOperationComplete. I ended up adopting that suggestion because I think it is good.

In this case, I kept things the way they are because I think there is an advantage to ending all attempts at the same time as it makes the results more consistent. I see the argument for not calling onAttemptComplete for the successful calls until onOperationComplete is called as being a more accurate measurement, but making this change wouldn't improve the accuracy of the failed attempts so I'm not sure it's worth it unless there is some way to make the measurements more accurate for all attempts.